controllers/resource/template.go

```diff
@@ -61,6 +62,42 @@ type NutanixTemplate struct {
 	now anywhereTypes.NowFunc
 }
 
+func GetVSphereCredValues(credSecret *corev1.Secret) (map[string]string, error) {
```

Comment:
does this need to be public?

Reply:
It doesn't...but the existing public methods on this package are very difficult to test. The [test cases](https://github.com/aws/eks-anywhere/blob/808b159fa444a67fdab675e90d55ea4c93cb196b/controllers/resource/reconciler_test.go#L112) involve ~100 lines of setup each. Figuring all of that out to test ~6 lines of code in order to make this method private (which it probably should be) just didn't feel worth it.

Reply:
Got you
then I would suggest moving this method to `controllers/resource/internal/vsphere.go`
That way we can test it individually but it's not exposed to other modules/packages

Reply:
That's perfect - thanks!